ngraph/test/type_prop/squeeze.cpp

```diff
@@ -40,11 +55,41 @@ TEST(type_prop, squeeze_dynamic)
     axes_node = make_shared<ngraph::op::Constant>(element::u64, Shape{0}, vector<int64_t>{});
     auto squeeze_default_axes = make_shared<op::Squeeze>(param, axes_node);
 
+    ASSERT_EQ(squeeze_default_axes->get_element_type(), element::f32);
+    EXPECT_TRUE(
+        squeeze_default_axes->get_output_partial_shape(0).same_scheme(PartialShape::dynamic(6)));
+}
+
+TEST(type_prop, squeeze_dynamic_dynamic_rank)
+{
+    auto param = make_shared<op::Parameter>(element::f32, PartialShape::dynamic());
+    auto axes_node =
+        make_shared<ngraph::op::Constant>(element::u64, Shape{2}, vector<int64_t>{0, 2});
+    auto squeeze = make_shared<op::Squeeze>(param, axes_node);
+
+    ASSERT_EQ(squeeze->get_element_type(), element::f32);
+
+    EXPECT_TRUE(squeeze->get_output_partial_shape(0).same_scheme(PartialShape::dynamic()));
+
+    axes_node = make_shared<ngraph::op::Constant>(element::u64, Shape{0}, vector<int64_t>{});
+    auto squeeze_default_axes = make_shared<op::Squeeze>(param, axes_node);
+
     ASSERT_EQ(squeeze_default_axes->get_element_type(), element::f32);
     EXPECT_TRUE(
         squeeze_default_axes->get_output_partial_shape(0).same_scheme(PartialShape::dynamic()));
 }
 
+TEST(type_prop, squeeze_axes_dynamic)
+{
+    auto param = make_shared<op::Parameter>(element::f32, Shape{1, 4, 1, 4, 1, 8});
+    auto axes_node =
+        make_shared<ngraph::op::Parameter>(element::u64, PartialShape::dynamic());
+    auto squeeze = make_shared<op::Squeeze>(param, axes_node);
+
+    ASSERT_EQ(squeeze->get_element_type(), element::f32);
+    ASSERT_TRUE(squeeze->get_output_partial_shape(0).same_scheme(PartialShape::dynamic(6)));
```

Comment:
Does it mean that the output shape has static `rank` filled with fully dynamic dimensions?
But axes node is Parameter and first input shape contains few `1` so it can be squeezed and output will have smaller rank.